### PR TITLE
feat(storage): add bytesRange option for downloadData

### PR DESCRIPTION
**Pull request by HuiSF**
#### Description of changes

* Reverted the changes of exposing the `defaultPrefixResolver`
* Added `bytesRange` option to the `downloadData` API

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/storage/src/providers/s3/types/options.ts [hunk 1]
```diff
@@ -26,6 +26,13 @@ type WriteOptions = {
 	accessLevel?: StorageAccessLevel;
 };
 
+type ByteRangeOptions = {
```
Comment (AllanZhengYP) [inline packages/storage/src/providers/s3/types/options.ts:29]: Nit:
```suggestion
type BytesRangeOptions = {
```

File: packages/storage/src/providers/s3/types/options.ts [hunk 2]
```diff
@@ -80,7 +87,10 @@ export type GetUrlOptions = ReadOptions &
 /**
  * Input options type for S3 downloadData API.
  */
-export type DownloadDataOptions = ReadOptions & CommonOptions & TransferOptions;
+export type DownloadDataOptions = ReadOptions &
+	CommonOptions &
+	TransferOptions &
+	ByteRangeOptions;
```
Comment (AllanZhengYP) [inline packages/storage/src/providers/s3/types/options.ts:93]: Ditto
```suggestion
	BytesRangeOptions;
```

**Comment by AllanZhengYP** [comment]
Looks good. only have 1 nit

**Review by kvramyasri7 — APPROVED** [review]
LGTM. Thanks Hui.

**Review by ashwinkumar6 — APPROVED** [review]
lgtm
Thank you 💯

**Pull request merged**